### PR TITLE
BROKERS: POST Image Generation

### LINKED ISSUE #204: BROKERS: POST Image Generation
# The Ask
We need to develop an OpenAI Broker Method that POSTs text and returns images from OpenAI API.

# How to Complete this Task
Here's some steps to complete this deliverable.

- Ensure that you understand how to natively talk to OpenAI API using Postman - (includes account & API key)
- Find the reference to integrating with OpenAI APIs [here](https://platform.openai.com/docs/api-reference/images/create)
- Learn what Brokers are [here](https://github.com/hassanhabib/The-Standard/blob/master/1.%20Brokers/1.%20Brokers.md#1-brokers)
- Watch how The Standard Community developed a similar Broker for OpenAI [here](https://www.youtube.com/watch?v=YRLymNBSlqI&list=PLan3SCnsISTTl_MnGP6B78Nfu9Ix8G4mU&index=11)
- If you have any questions - ask The Standard Community [here](https://discord.com/invite/vdPZ7hS52X)

# How to claim this issue?
- Post on this issue a comment that says: "I claim this issue"
- You will be given 24 hours to complete the issue - if no PR is opened within 24 hours someone else will claim it
- If you have an open PR but is inactive for more than 24 hours from the last review comment we will shutdown your PR

### PR DESCRIPTION
**Pull request by Gummimundur**
Closes #204

### CODE REVIEW

File: Standard.AI.OpenAI/Models/Services/Foundations/ExternalImageGenerations/ExternalImageGenerationRequest.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+namespace Standard.AI.OpenAI.Models.Services.Foundations.ExternalImageGenerations
+{
+    internal class ExternalImageGenerationRequest
+    {
+        [JsonProperty(propertyName: "prompt")]
+        public string Prompt { get; set; }
+
+        [JsonProperty(propertyName: "n")]
+        public int ImagesToGenerate { get; set; } = 1;
```
Comment (BrianLParker) [inline Standard.AI.OpenAI/Models/Services/Foundations/ExternalImageGenerations/ExternalImageGenerationRequest.cs:16]: Remove Default

File: Standard.AI.OpenAI/Models/Services/Foundations/ExternalImageGenerations/ExternalImageGenerationRequest.cs [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+        [JsonProperty(propertyName: "size")]
+        public string ImageSize { get; set; } = "1024x1024";
```
Comment (BrianLParker) [inline Standard.AI.OpenAI/Models/Services/Foundations/ExternalImageGenerations/ExternalImageGenerationRequest.cs:19]: Remove default value.

File: Standard.AI.OpenAI/Models/Services/Foundations/ExternalImageGenerations/ExternalImageGenerationRequest.cs [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+        [JsonProperty(propertyName: "response_format")]
+        public string ResponseFormat { get; set; } = "url";
```
Comment (BrianLParker) [inline Standard.AI.OpenAI/Models/Services/Foundations/ExternalImageGenerations/ExternalImageGenerationRequest.cs:22]: Remove default value.

File: Standard.AI.OpenAI/Models/Services/Foundations/ExternalImageGenerations/ExternalImageGenerationRequest.cs [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+
+        [JsonProperty(propertyName: "user")]
+        public string User { get; set; } = String.Empty;
```
Comment (BrianLParker) [inline Standard.AI.OpenAI/Models/Services/Foundations/ExternalImageGenerations/ExternalImageGenerationRequest.cs:25]: Remove default value.

**Review by BrianLParker — CHANGES_REQUESTED** [review]
Default values are handled by OpenAI.
RESTFulSense has been updated to prevent default values from being posted with the `ignoreDefaultValues` option to prevent over posting.
When the first broker was created this was not the case. An Issue should be raised to remove them as well.

**Comment by Gummimundur** [comment]
@BrianLParker Oh yeah of course. Thanks for the clarification.

**Review by BrianLParker — APPROVED** [review]
I'm good with this. Good catch on the optional `ExternalImageGenerationResult` properties.

**Review by ElbekDeveloper — APPROVED** [review]

**Comment by ElbekDeveloper** [comment]
LGTM

**Comment by BrianLParker** [comment]
We are going to as for screen shots of both:

1. Postman testing as required.
2. The broker running in a temp test as we do in the video's

File: Standard.AI.OpenAI/Models/Services/Foundations/ExternalImageGenerations/ExternalImageGenerationResponse.cs [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+namespace Standard.AI.OpenAI.Models.Services.Foundations.ExternalImageGenerations
+{
+    internal class ExternalImageGenerationResponse
+    {
+        [JsonProperty(propertyName: "created")]
+        public int Created { get; set; }
+
+        [JsonProperty(propertyName: "data")]
+        public ExternalImageGenerationResult[] GeneratedImages { get; set; }
```
[the remark this answers is not in the document]
Comment (Gummimundur) [reply Standard.AI.OpenAI/Models/Services/Foundations/ExternalImageGenerations/ExternalImageGenerationResponse.cs:15]: Sounds good to me

**Review by BrianLParker — APPROVED** [review]

**Pull request merged**